File: clojure/hello-world/src/hello_world.clj
(ns hello-world)

(def defines-some "variable")

;; (def hello (fn [name] "Hello"))
(defn hello [name] "Hello")

;; sdfasdf
;; (fn [argument1 argument2] "pizza")
;; function(arg1, arg2) { return "pizza"; }
(defn hello
  ([] (hello "World"))
  ([arg] (str "Hello, " arg "!")))

File: java/hello-world/src/main/java/HelloWorld.java
public class HelloWorld {
    public static String hello(String name) {
        if (name == null) {
            return "Hello, World!";
        } else if (name == "") {
            return "Hello, World!";
        }
        else {
            return "Hello, " + name + "!";
        }
    }
}
